### PR TITLE
refactor: Rename Python dependency filenames

### LINKED ISSUE #4994: Can we merge requirement*.txt files from three down to two?
I am always frustrated when I am looking to update a Python dependency that I need to look at three requirements files instead of just two.

### Describe the problem that you'd like solved

This repo currently has:
1. requirements.txt
2. requirements_common.txt
3. requirements_test.txt

Would it be a good idea to simplify and merge requirements so that we just have:
1. requirements.txt
2. requirements_test.txt  # imports requirements.txt to avoid errors.

Or is there some logic to having three files?

### Stakeholders

@dhruvmanila

### PR DESCRIPTION
**Pull request by dhruvmanila**
Second PR to move towards simpler dependency files:
- Renamed `requirements_common.txt` to `requirements.txt`
- Update dependency filenames in `Dockerfile.olbase` and `requirements_test.txt`

Closes #4994 

### Stakeholders

@cdrini

### CODE REVIEW

File: requirements_test.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 # requirements for running tests
 # Expects some apt packages are already installed.
-# see .travis.yml for more details.
+# see .github/workflows/python_tests.yml for more details.
 
--r requirements_common.txt
+-r requirements.txt
 debugpy>=1.2.0
```
Comment (cclauss) [inline requirements_test.txt:6]: [`debugpy`](https://github.com/microsoft/debugpy) is not required in our GitHub Actions but we need it to be in the Docker container if we want to https://github.com/internetarchive/openlibrary/wiki/Debugging

Comment (dhruvmanila) [reply requirements_test.txt:6]: Right, so this should be moved to `requirements.txt` instead.

**Review by cclauss — APPROVED** [review]
LGTM

File: requirements.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@ amightygirl.paapi5-python-sdk==1.0.0
 Babel==2.9.0
 beautifulsoup4==4.9.3
 DBUtils==1.3
+debugpy>=1.2.0
```
Comment (cdrini) [inline requirements.txt:5]: Oooh, actually I think we do want to keep this in requirements_test.txt. requirements_test.txt is basically any non-production necessary requirements. Let's update the comment in requirements_test.txt to note that it's not just GHA specific.

**Review by cdrini — CHANGES_REQUESTED** [review]
Lgtm! One small change.

File: requirements.txt (hunk 2 above)
Comment (cclauss) [reply requirements.txt:5]: If we do so, we need to separate aways to build our Docker images:
1. A production use Docker build that includes debugpy
2. A non-production use Docker build that does not include debugpy.

Could there be a more streamlined process so to combine these two build types?

Comment (cdrini) [reply requirements.txt:5]: (I think you have those flipped; production docker build WITHOUT debugpy; and non-prod build WITH debugpy). 

olbase/oldev were meant to be that; but we currently use `oldev` for production (which bring in all the testing stuff). We should at some point transition to using only olbase on production, so we don't have any of that.

Comment (cdrini) [reply requirements.txt:5]: (That way we'll be able to remove the `build` step during our deploys entirely! Which should speed things up :)

Comment (dhruvmanila) [reply requirements.txt:5]: Sounds good!

File: requirements_test.txt [hunk 3]
```diff
@@ -1,8 +1,9 @@
-# requirements for running tests
+# Requirements for running tests and non-production Docker build.
 # Expects some apt packages are already installed.
 # see .github/workflows/python_tests.yml for more details.
 
 -r requirements.txt
+debugpy>=1.2.0
 flake8==3.9.0
 mypy==0.812
 pymemcache==3.4.1
```
Comment (cclauss) [inline requirements_test.txt]: Is `pymemcache` _only_ used for testing?

Comment (cdrini) [reply requirements_test.txt]: I think sorta; during testing, if the memcache servers aren't running it "falls back" to pymemcache as an alternative.

**Review by cdrini — APPROVED** [review]
Lgtm! Staging started up ok.

**Pull request merged**